pjsip/src/pjsip/sip_transport_tcp.c

```diff
@@ -1396,6 +1396,18 @@ static pj_bool_t on_data_read(pj_activesock_t *asock,
             pjsip_tpmgr_receive_packet(rdata->tp_info.transport->tpmgr,
                                        rdata);
 
+        if (size_eaten == -1) {
```

Comment:
Generally I'd prefer to see `if (size_eaten < 0)` to make it clear that negative responses indicate errors (rather than `-1` being a magic number).